File: app/src/main/java/com/kota/Bahamut/Command/BahamutCommandFSendMail.java
package com.kota.Bahamut.Command;

import static com.kota.Bahamut.Service.CommonFunctions.getContextString;

import com.kota.ASFramework.Dialog.ASProcessingDialog;
import com.kota.ASFramework.UI.ASToast;
import com.kota.Bahamut.ListPage.TelnetListPage;
import com.kota.Bahamut.ListPage.TelnetListPageBlock;
import com.kota.Bahamut.R;
import com.kota.Telnet.Reference.TelnetKeyboard;
import com.kota.Telnet.TelnetClient;
import com.kota.Telnet.TelnetOutputBuilder;

public class BahamutCommandFSendMail extends TelnetCommand  {
    String _receiver;
    public BahamutCommandFSendMail(String receiver) {
        this._receiver = receiver;
        this.Action = SendMail;
    }

    @Override
    public void execute(TelnetListPage telnetListPage) {
        if (_receiver.length() > 0) {
            TelnetClient.getClient().sendDataToServer(
                    TelnetOutputBuilder.create()
                            .pushString("FA\n")
                            .pushKey(TelnetKeyboard.CTRL_Y)
                            .pushString(_receiver + "\n")
                            .build());
        }
        ASProcessingDialog.showProcessingDialog(getContextString(R.string.board_page_send_mail_ing));
    }

    @Override
    public void executeFinished(TelnetListPage telnetListPage, TelnetListPageBlock telnetListPageBlock) {
        setDone(true);
        ASProcessingDialog.dismissProcessingDialog();
        ASToast.showShortToast(getContextString(R.string.board_page_send_mail_finish));
    }

    public String toString() {
        return "[FSendMail]";
    }
}

File: app/src/main/java/com/kota/Bahamut/command/BahamutCommandFSendMail.kt
package com.kota.Bahamut.command

import com.kota.asFramework.dialog.ASProcessingDialog.Companion.dismissProcessingDialog
import com.kota.asFramework.dialog.ASProcessingDialog.Companion.showProcessingDialog
import com.kota.asFramework.ui.ASToast.showShortToast
import com.kota.Bahamut.listPage.TelnetListPage
import com.kota.Bahamut.listPage.TelnetListPageBlock
import com.kota.Bahamut.R
import com.kota.Bahamut.service.CommonFunctions.getContextString
import com.kota.telnet.reference.TelnetKeyboard
import com.kota.telnet.TelnetClient
import com.kota.telnet.TelnetOutputBuilder.Companion.create

class BahamutCommandFSendMail(var receiver: String) : TelnetCommand() {
    init {
        this.action = BahamutCommandDef.Companion.SEND_MAIL
    }

    override fun execute(telnetListPage: TelnetListPage) {
        if (receiver.isNotEmpty()) {
            TelnetClient.myInstance!!.sendDataToServer(
                create()
                    .pushString("FA\n")
                    .pushKey(TelnetKeyboard.CTRL_Y)
                    .pushString(receiver + "\n")
                    .build()
            )
        }
        showProcessingDialog(getContextString(R.string.board_page_send_mail_ing))
    }

    override fun executeFinished(
        telnetListPage: TelnetListPage,
        telnetListPageBlock: TelnetListPageBlock?
    ) {
        isDone = true
        dismissProcessingDialog()
        showShortToast(getContextString(R.string.board_page_send_mail_finish))
    }

    override fun toString(): String {
        return "[FSendMail]"
    }
}
